### PR TITLE
Improve AndroidView resize jank.

### LINKED ISSUE #19572: Reduce visual artifacts when resizing embedded Android views
Right now a bad artifact can be seen when an AndrodView is resized.
The root cause is that the render tree changes (happening on the ui thread) are not synchronized with the Android view resize (happening on the platform thread) so we end up rendering with a scaled frame of the Android view.
This bug is not about solving the synchronization issue (@jason-simmons might have some ideas around that), but about reducing the severity of the resize artifact.

The current plan is to detect the inconsistent state at paint time (on the gpu thread) and do something more graceful than just scaling.

### PR DESCRIPTION
**Pull request by amirh**
Resizing an AndroidView happens asynchronously (as it requires thread
hopping from the ui thread to the platform thread).
While waiting for the resize to complete the framework does exactly when
the embedded view has been resized. This leads to pretty bad jank as the
framework might paint the embedded view with a wrong size.

We're working around this by freezing the texture frame while resizing
until we are sure that the next frame has the new size. This is how it
looks with the workaround:

This is how it looks before and after the workaround:

Before (janky) |  After (less janky)
:--------|---------:
resize_before | resize_after

Fixes #19572

### CODE REVIEW

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 1]
```diff
@@ -91,9 +96,30 @@ class RenderAndroidView extends RenderBox {
...
+      context.pushClipRect(true, offset, offset & size, _paintTexture);
+     return;
+    }
+
+    _paintTexture(context, offset);
+  }
+
+  void _paintTexture(PaintingContext context, Offset offset) {
+    // As resizing the Android view happens asynchronously we don't know exactly when is a
+    // texture frame with the new size ready for consumption.
```
Comment (jason-simmons) [inline packages/flutter/lib/src/rendering/platform_view.dart]: typo: "size is ready"

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 2]
```diff
@@ -91,9 +96,30 @@ class RenderAndroidView extends RenderBox {
     if (_viewController.textureId == null)
       return;
 
+    // Clip the texture if it's going to paint out of the bounds of the renter box
+    // (see comment in _paintTexture for an explanation of when this happens).
+    if (size.width < _currentAndroidViewSize.width || size.height < _currentAndroidViewSize.height) {
+      context.pushClipRect(true, offset, offset & size, _paintTexture);
+     return;
```
Comment (jason-simmons) [inline packages/flutter/lib/src/rendering/platform_view.dart]: style: align this with the line above

**Review by jason-simmons — COMMENTED** [review]
LGTM

File: packages/flutter/lib/src/rendering/platform_view.dart (hunk 2 above)
Comment (amirh) [reply packages/flutter/lib/src/rendering/platform_view.dart]: done

File: packages/flutter/lib/src/rendering/platform_view.dart (hunk 1 above)
Comment (amirh) [reply packages/flutter/lib/src/rendering/platform_view.dart]: done

File: packages/flutter/lib/src/rendering/layer.dart [hunk 3]
```diff
@@ -224,6 +225,8 @@ class TextureLayer extends Layer {
   /// The identity of the backend texture.
   final int textureId;
 
+  final bool freeze;
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/layer.dart]: can you add some dartdocs to this?

Comment (amirh) [reply packages/flutter/lib/src/rendering/layer.dart]: sent #20383

**Pull request merged**